runtime/src/main/java/org/corfudb/runtime/clients/NettyClientRouter.java

```diff
@@ -361,6 +362,9 @@ private synchronized void connectChannel(Bootstrap b)
                         log.warn("Exception while reconnecting, retry in {} ms", timeoutRetry, e);
                         Sleep.MILLISECONDS.sleepUninterruptibly(timeoutRetry);
                     }
+                    if (--retryCount == 0) {
+                        throw new NetworkException("Connection retry limit reached.", node);
```

Comment:
This is not a good idea. No one will see this exception since it runs on a Netty future.

Reply:
This code was throwing an un-handled exception earlier after a certain number of retries.
```
2018-01-17 15:49:49,899 WARN  [client-1] i.n.u.c.DefaultPromise - An exception was thrown by org.corfudb.runtime.clients.NettyClientRouter$$Lambda$123/1147741970.operationComplete()
 org.corfudb.runtime.exceptions.NetworkException: Retry limit reached. [endpoint=tcp://localhost:9001/]
	at org.corfudb.runtime.clients.NettyClientRouter.lambda$connectChannel$2(NettyClientRouter.java:359)
```